ext/sockets/sockets.stub.php

```diff
@@ -24,65 +24,65 @@ function socket_listen(Socket $socket, int $backlog = 0): bool {}
 
 function socket_close(Socket $socket): void {}
 
-function socket_write(Socket $socket, string $buf, ?int $length = null): int|false {}
+function socket_write(Socket $socket, string $data, ?int $max_length = null): int|false {}
 
-function socket_read(Socket $socket, int $length, int $type = PHP_BINARY_READ): string|false {}
+function socket_read(Socket $socket, int $length, int $mode = PHP_BINARY_READ): string|false {}
 
 /**
- * @param string $addr
+ * @param string $address
  * @param int $port
  */
-function socket_getsockname(Socket $socket, &$addr, &$port = null): bool {}
+function socket_getsockname(Socket $socket, &$address, &$port = null): bool {}
 
 /**
- * @param string $addr
+ * @param string $address
  * @param int $port
  */
-function socket_getpeername(Socket $socket, &$addr, &$port = null): bool {}
+function socket_getpeername(Socket $socket, &$address, &$port = null): bool {}
 
 function socket_create(int $domain, int $type, int $protocol): Socket|false {}
 
-function socket_connect(Socket $socket, string $addr, ?int $port = null): bool {}
+function socket_connect(Socket $socket, string $address, ?int $port = null): bool {}
 
-function socket_strerror(int $errno): string {}
+function socket_strerror(int $error_code): string {}
 
-function socket_bind(Socket $socket, string $addr, int $port = 0): bool {}
+function socket_bind(Socket $socket, string $address, int $port = 0): bool {}
 
-/** @param string|null $buf */
-function socket_recv(Socket $socket, &$buf, int $len, int $flags): int|false {}
+/** @param string|null $data */
+function socket_recv(Socket $socket, &$data, int $length, int $flags): int|false {}
```

Comment:
I was wondering before if our `$length` and `$max_length` usages are really consistent. E.g. the manual says here `Up to len bytes will be fetched from remote host` so it seems it could also be `$max_length`.

Reply:
I've been wondering about this as well. I think it might make sense to make all of these use just `$length`, unless they have some non-standard meaning (e.g. in the zlib case, where "max length" refers to a limit on the decoded payload size).

Reply:
Yes, it's what I also though about.

Reply:
I've done this now. Everything uses just $length now, apart from zlib uncompression:
```
ext/zlib/zlib.stub.php:function zlib_decode(string $data, int $max_length = 0): string|false {}
ext/zlib/zlib.stub.php:function gzinflate(string $data, int $max_length = 0): string|false {}
ext/zlib/zlib.stub.php:function gzdecode(string $data, int $max_length = 0): string|false {}
ext/zlib/zlib.stub.php:function gzuncompress(string $data, int $max_length = 0): string|false {}
```